src/kOS/Suffixed/PartModuleField/PartModuleFields.cs

```diff
@@ -133,49 +122,40 @@ private bool IsLegalValue(BaseField field, ref Structure newVal, out KOSExceptio
                     return false;
                 }
             }
-            List<UI_Control> controls = GetFieldControls(field);
 
-            // It's really normal for there to be only one control on a KSPField, but because
-            // it's technically possible to have more than one according to the structure of
-            // the API, this loop is here to check all of "them":
-            foreach (UI_Control control in controls)
+            // Some of these are subclasses of each other, so don't change this to an if/else.
+            // It's a series of if's on purpose so it checks all classes the control is derived from.
+            if (control is UI_Toggle)
             {
-                // Some of these are subclasses of each other, so don't change this to an if/else.
-                // It's a series of if's on purpose so it checks all classes the control is derived from.
-                if (control is UI_Toggle)
-                {
-                    // Seems there's nothing to check here, but maybe later there will be?
-                }
-                if (control is UI_Label)
-                {
-                    except = new KOSInvalidFieldValueException("Labels are read-only objects that can't be changed");
-                    isLegal = false;
-                }
-                var vector2 = control as UI_Vector2;
-                if (vector2 != null)
+                // Seems there's nothing to check here, but maybe later there will be?
+            }
+            if (control is UI_Label)
+            {
+                except = new KOSInvalidFieldValueException("Labels are read-only objects that can't be changed");
+                isLegal = false;
+            }
+            var vector2 = control as UI_Vector2;
+            if (vector2 != null)
+            {
+                // I have no clue what this actually looks like in the UI?  What is a
+                // user editable 2-D vector widget?  I've never seen this before.
+                if (convertedVal != null)
                 {
-                    // I have no clue what this actually looks like in the UI?  What is a
-                    // user editable 2-D vector widget?  I've never seen this before.
-                    if (convertedVal != null)
```

Comment:
The above looks like a large edit, but if you look carefully, all it is is just a change of indent level.